lib/fluent/plugin/in_unix.rb

```diff
@@ -80,22 +98,22 @@ def run
     # }
     def on_message(msg)
       # TODO format error
-      tag = msg[0].to_s
+      tag = @tag || (msg[0].to_s)
       entries = msg[1]
 
-      if entries.class == String
+      case entries
+      when String
         # PackedForward
-        es = MessagePackEventStream.new(entries)
+        es = Fluent::MessagePackEventStream.new(entries)
         router.emit_stream(tag, es)
 
-      elsif entries.class == Array
+      when Array
         # Forward
-        es = MultiEventStream.new
+        es = Fluent::MultiEventStream.new
         entries.each {|e|
           record = e[1]
           next if record.nil?
-          time = e[0]
-          time = (now ||= EventTime.now) if time.to_i == 0
+          time = convert_time(e[0])
```

Comment:
since in #convert_time check `time == 0`,  it's `e[0].to_i` ?

Reply:
No. v1 supports nanosecond, so Float and EventTime are also supported.
`== 0` works with these types.

Reply:
if `e[0]` is `nil`(I'm not sure this happens), should we do it?